src/cluster_legacy.c

```diff
@@ -6278,13 +6278,26 @@ const char *clusterGetMessageTypeString(int type) {
     return "unknown";
 }
 
-/* Get the slot from robj and return it. If the slot is not valid,
- * return -1 and send an error to the client. */
-int getSlotOrReply(client *c, robj *o) {
+int getSlotOrError(robj *o, char **err_out) {
```

Comment:
I think the functions getSlotOrError and getSlotOrReply are overlapped in some logic, I prefre to combine them as getSlotOrReply(client *c, robj *o, char* message)

Reply:
It works like this way? 

int getSlotOrReply(client *c, robj *o, char **err_out) {
    long long slot;

    if (getLongLongFromObject(o, &slot) != C_OK || slot < 0 || slot >= CLUSTER_SLOTS) {
        *err_out = "Invalid or out of range slot";
         addReplyErrorSds(c, sdsnew(err));
         return -1;
    }

    return (int)slot;
}

Reply:
In the case for this PR we want to parse the slot but not add a reply to the client (since we leave this to the actual command handler). This is why they are two functions (one does not reply to the client)